### PR TITLE
Add resource usage guidelines docs

### LINKED ISSUE #5950: Provide better guidance on agent resource requirements

### PR DESCRIPTION
**Pull request by thampiotr**
#### PR Description

Add resource usage guidelines based on our experiences.

#### Which issue(s) this PR fixes

Fixes #5950

#### Notes to the Reviewer

Note: the traces usage is too low to provide any reasonable guidelines. We will need to come back to this.

#### PR Checklist

- [x] Documentation added

### CODE REVIEW

File: docs/sources/flow/monitoring/agent-resource-usage.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+most users, but your actual usage may be different.
+
+{{% /admonition %}}
+
+## Prometheus metrics
+
+The Prometheus metrics resource usage depends mainly on the number of active series
+that need to be scraped.
+
+As a rule of thumb, **per each 1 million active series** with 15s scrape interval,
```
Comment (mattdurham) [inline docs/sources/flow/monitoring/agent-resource-usage.md:32]: We should use 60s here since that is what we recommend everywhere.

Comment (thampiotr) [reply docs/sources/flow/monitoring/agent-resource-usage.md:32]: Hmm, in that case we don't have a data from production... unless we change our prod to match what we recommend. Any idea why do we not use the recommended interval?

Comment (jcreixell) [reply docs/sources/flow/monitoring/agent-resource-usage.md:32]: AFAIK 60s is optimized for Grafana Cloud and is documented here https://grafana.com/docs/grafana-cloud/cost-management-and-billing/understand-your-invoice/metrics-invoice/#data-points-per-minute-dpm. It is possible to tweak it via scrape config but it has a direct impact on cost

Comment (thampiotr) [reply docs/sources/flow/monitoring/agent-resource-usage.md:32]: I guess the best way forward is for me to set up an experiment where we do a 60s scrape interval for some time to collect data?

Comment (jcreixell) [reply docs/sources/flow/monitoring/agent-resource-usage.md]: We can try and see what we get, I assume most OSS users will use the 15s interval and Cloud users would prefer the 60s interval. My guess is that memory usage should be similar, but CPU and memory will change (more data points to filter and send)

File: docs/sources/flow/monitoring/agent-resource-usage.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+The Prometheus metrics resource usage depends mainly on the number of active
+series that need to be scraped.
+
+As a rule of thumb, **per each 1 million active series** 
+you can expect to use approximately:
+
+* 1.5 CPU cores
+* 11 GiB of memory
+* 1.5 MiB/s of total network bandwidth, send and receive
```
Comment (thampiotr) [inline docs/sources/flow/monitoring/agent-resource-usage.md:38]: @mattdurham @jcreixell - this is based on experimental data from setting scrape interval to 1m. I think it's the best data we have right now until we start doing benchmarks more systematically and I'd recommend we go with this for now. LMK if you have any objections :)

File: docs/sources/flow/monitoring/agent-resource-usage.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+The resource usage depends on the workload, hardware and the configuration used.
+The information on this page is a good starting point for most users, but your
+actual usage may be different.
+
+{{% /admonition %}}
+
+## Prometheus metrics
+
+The Prometheus metrics resource usage depends mainly on the number of active
+series that need to be scraped.
```
Comment (jcreixell) [inline docs/sources/flow/monitoring/agent-resource-usage.md:31]: nit: The Prometheus metrics resource usage depends mainly on the number of active
series that need to be scraped **, and the scrape interval**.

File: docs/sources/flow/monitoring/agent-resource-usage.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+As a rule of thumb, **per each 1 million active series**
```
Comment (jcreixell) [inline docs/sources/flow/monitoring/agent-resource-usage.md:33]: nit: mention here "with default settings"

**Review by jcreixell — APPROVED** [review]
added some nits, LGTM overall

**Pull request merged**